docs/api/spec/query-api.yaml

```diff
@@ -0,0 +1,288 @@
+        content:
+          application/json:
+            example:
+              name: 'trees'
+        required: true
+      responses:
+        '201':
+          description: 'Successfully create new type'
+          content:
+            application/json:
+              example:
+                550e8400-e29b-41d4-a716-446655440019: 'trees'
+        '401':
+          $ref: '#/components/responses/UnauthorizedError'
+      deprecated: false
+  '/types/{type_uuid}':
+    patch:
+      tags:
+        - Types info and manipulation
+      summary: Modify existing type in the database
+      description: 'Modify exisiting type in the database'
+      security:
+        - OAuth2: [admin]
+      parameters:
+        - name: type_uuid
+          description: 'ID of specific token to retrieve'
+          in: path
+          required: true
+          schema:
+            type: string
+            format: uuid
+            example: '550e8400-e29b-41d4-a716-446655440000'
+      requestBody:
+        description: 'Name of type'
+        content:
+          application/json:
+            example:
+              name: 'above_the_ground_trees'
+        required: true
+      responses:
+        '200':
+          description: 'Successfully modified the type'
+          content:
+            application/json:
+              example:
+                550e8400-e29b-41d4-a716-446655440019: 'above_the_ground_trees'
+        '401':
+          $ref: '#/components/responses/UnauthorizedError'
+        '404':
+          $ref: '#/components/responses/NotfoundError'
+      deprecated: false
+
+  '/user/:user_id/types/:type_id':
+    get:
+      tags:
+        - Get user like / unlike activities on a type
+      summary: Get user like / unlike activities
+      description: 'Get user like / unlike activities, this api is open for everyone, so no token and access control needed'
+      security: []
+      responses:
+        '200':
+          description: 'Successfully retrieve user activities'
+          content:
+            application/json:
+              example:
+                550e8400-e29b-41d4-a716-446655440001: [550e8400-e29b-41d4-a716-446655440024,550e8400-e29b-41d4-a716-446655440029]
+                550e8400-e29b-41d4-a716-446655440010: [550e8400-e29b-41d4-a716-446655440019, 550e8400-e29b-41d4-a716-446655440025]
+        '401':
+          $ref: '#/components/responses/UnauthorizedError'
+        '404':
+          $ref: '#/components/responses/NotfoundError'
+      deprecated: false
+
+components:
+  securitySchemes:
+    OAuth2:
+      type: oauth2
+      flows:
+        authorizationCode:
+          authorizationUrl: https://dev-k8s.treetracker.org/keycloak/realm/treetracker/oauth/authorize
+          tokenUrl: https://dev-k8s.treetracker.org/keycloak/realm/treetracker/oauth/token
+          scopes:
+            read: Grants read access
+            write: Grants write access
+            admin: Grants access to admin operations
```

Comment:
The oauth2, we have deployed the keycloak to dev env, so we can use the url above to login, get/refresh token.